src/components/FeaturePanel/Climbing/CragsInAreaFilter.tsx

```diff
@@ -0,0 +1,224 @@
+import {
+  Badge,
+  Button,
+  IconButton,
+  ListSubheader,
+  Menu,
+  Slider,
+  Stack,
+  Tooltip,
+} from '@mui/material';
+import React, { useEffect, useMemo } from 'react';
+import FilterListAltIcon from '@mui/icons-material/FilterListAlt';
+import { t } from '../../../services/intl';
+import { GRADE_TABLE } from './utils/grades/gradeData';
+import {
+  convertGrade,
+  getDifficulty,
+  isInGradeInterval,
+} from './utils/grades/routeGrade';
+import { useUserSettingsContext } from '../../utils/UserSettingsContext';
+import { RouteDifficultyBadge } from './RouteDifficultyBadge';
+import { GradeSystemSelect } from './GradeSystemSelect';
+
+export const useCragsInAreaFilter = () => {
+  const [gradeInterval, setGradeInterval] = React.useState<number[] | null>(
+    null,
+  );
+  const [minimumRoutesInInterval, setMinimumRoutesInInterval] =
+    React.useState<number>(1);
+  const [isTouched, setIsTouched] = React.useState<boolean>(false);
+  const { userSettings } = useUserSettingsContext();
+  const currentGradeSystem = userSettings['climbing.gradeSystem'] || 'uiaa';
+  const values = GRADE_TABLE[currentGradeSystem];
+  const uniqueValues = useMemo(() => [...new Set(values)], [values]);
+
+  return {
+    uniqueValues,
+    gradeInterval,
+    setGradeInterval,
+    minimumRoutesInInterval,
+    setMinimumRoutesInInterval,
+    setIsTouched,
+    isTouched,
+  };
+};
+
+export const filterCrag =
+  ({
+    gradeInterval,
+    currentGradeSystem,
+    uniqueValues,
+    minimumRoutesInInterval,
+  }) =>
+  (crag) => {
+    if (!gradeInterval) return true;
+
+    const numberOfFilteredRoutes = crag.memberFeatures.reduce((acc, route) => {
+      const difficulty = getDifficulty(route.tags);
+      if (!difficulty) return acc;
+      const convertedGrade = convertGrade(
+        difficulty.gradeSystem,
+        currentGradeSystem,
+        difficulty.grade,
+      );
+      if (
+        isInGradeInterval({
+          gradeMin: uniqueValues[gradeInterval[0]],
+          gradeMax: uniqueValues[gradeInterval[1]],
+          grade: convertedGrade,
+          currentGradeSystem,
+        })
+      )
+        return acc + 1;
+      return acc;
+    }, 0);
+    return numberOfFilteredRoutes > minimumRoutesInInterval;
+  };
+
+export const CragsInAreaFilter = ({
+  gradeInterval,
+  setGradeInterval,
+  minimumRoutesInInterval,
+  setMinimumRoutesInInterval,
+  uniqueValues,
+  setIsTouched,
+  isTouched,
+}) => {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+
+  const open = Boolean(anchorEl);
+  const { userSettings, setUserSetting } = useUserSettingsContext();
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleIfTouched = ({
```

Comment:
We could probably also extract/bundle some of these callbacks in the render function. But lets perhaps decide that after the other stuff is done. I'd be glad to pair code on this.